dio/lib/src/dio_event_processor.dart

```diff
@@ -87,8 +89,76 @@ class DioEventProcessor implements EventProcessor {
 
     return SentryResponse(
       headers: _options.sendDefaultPii ? headers : null,
-      bodySize: dioError.response?.data?.length as int?,
+      bodySize: _getBodySize(
+        dioError.response?.data,
+        dioError.requestOptions.responseType,
+      ),
       statusCode: response?.statusCode,
+      data: _getResponseData(
+        dioError.response?.data,
+        dioError.requestOptions.responseType,
+      ),
     );
   }
+
+  /// Returns the response data, if possible according to the users settings.
+  Object? _getResponseData(Object? data, ResponseType responseType) {
+    if (!_options.sendDefaultPii) {
+      return null;
+    }
+    if (data == null) {
+      return null;
+    }
+    switch (responseType) {
+      case ResponseType.json:
+        final js = json.encode(data);
```

Comment:
I wonder if `data` isn't always a `String` here? Not sure, would need to check `dio` internals.

Reply:
It can be any valid json element. This is covered by the unit test where we go through all possibilities.

```dart
responsetype.json: [
    9001, 
    null,
    'string',
    true,
    ['list'],
    {'map-key': 'map-value'},
 ]
```